source/common/tcp_proxy/upstream.h

```diff
@@ -15,6 +15,8 @@
 namespace Envoy {
 namespace TcpProxy {
 
+constexpr absl::string_view DisableTunnelingFilterStateKey = "envoy.tcp_proxy.disable_tunneling";
```

Comment:
To match other uses of FilterState, I think it would be better to define a specific filter state object that should be set, instead of any object with the given name, similar to `class PerConnectionCluster : public StreamInfo::FilterState::Object` (from tcp_proxy.h), and give it a boolean value.

Reply:
@ggreenway Let me quote Lizan from [another PR I merged some time ago](https://github.com/envoyproxy/envoy/pull/25234), Lizan asked to use StringAccessor, quoting his reasoning:

> "You should still use the key name envoy.upstream.dynamic_host but the type of filter object should be a StringAccessor and I don't see why it needs to be a custom type. A string accessor is easier to get/set by other filters and for [logging and header substitution](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-filter-state)"

And quoting Kuat from this PR:

> I think in the past we used a string accessor with "reason" as value. That might be more informative than a boolean.

I personally I don't have a hard preference between these two options, they both makes sense to me, but I tend towards StringAccessor. What should we agree about?

Reply:
@ggreenway IMHO the type system for filter states is based on their key name, not the C++ type. We don't have any "reflection" yet based on the content, so it's all just a blob for consumption sites.

Reply:
@kyessenov I don't think that's true; most/all FilterState has specific types. Each key has a specific type, so that something like `connection.streamInfo().filterState()->getDataReadOnly<PerConnectionCluster>()` will work.

I'm fine with using a common type for that (maybe add `BoolAccessor`), but I think putting the value into an Object, instead of just depending on whether the key exists, is clearer.

Reply:
done